### ISSUE TITLE
Can't load the new version after EOL

### ISSUE THREAD

**Issue by Taulim** [flathub/org.freedesktop.Platform.VulkanLayer.MangoHud#1]
I'm sure that I'm missing something really simple and this will be a silly question, but there we go...

I received the EOL warning this morning and decided to uninstall this package and re-install the rebased one (org.freedesktop.Platform.VulkanLayer.MangoHud), however I cannot see how to load this package inside the Steam flatpak runtime.

Is there something I need to set or is it something still needed on the Freedesktop / Steam flatpak side?

**Comment by gasinvein** [comment]
Thanks for pointing this out, `MangoHud//21.08` should be available now.

**Comment by Taulim** [comment]
Thank you for the quick fix.
Indeed using the branch 21.08 it does mount the folder as expected and I can find the mangohud binary.

However the hud seems to not be working. I think mangohud is not on the path.
If I use `/usr/lib/extensions/vulkan/MangoHud/bin/mangohud` %command% the game launches but I still have no hud.

Commands I have tried on steam:

- `WINE_FULLSCREEN_FSR=1 gamemoderun mangohud %command%` (This does not launch the game)
- `MANGOHUD=1 WINE_FULLSCREEN_FSR=1 gamemoderun %command%` (No hud here)
- `WINE_FULLSCREEN_FSR=1 gamemoderun /usr/lib/extensions/vulkan/MangoHud/bin/mangohud %command%` (No hud here)

**Comment by Taulim** [comment]
No, I normally use Proton GE builds.
I'm using the version 6.16-1.

To test I switched to official proton 6.3-6, deleted the compatdata folder so it could rebuild it.
But the problem is still present.

Some extra information.
O.S.: Pop_OS 21.04 nVidia
Flatpak: 1.11.2
MangoHud settings is on `/.var/app/com.valvesoftware.Steam/.config/MangoHud/`
Most of my system is standard, no crazy changes :smile: 

If it help at all I'm testing with `The Witcher 3`, mainly because it has good results with FSR (not the most demanding title though...).

**Comment by Taulim** [comment]
> It's normal Proton-GE, not flatpak one, right?

Yes, since flatpak 1.11.1 I switched to the normal :)



> Try running Steam via

This command-line worked (I'm setting the `MANGOHUD=1` per game basis)

**Comment by gasinvein** [comment]
So, the issue here is that pressure-vessel doesn't pick up vulkan layers from the new extension mount point (unless directed to it via `XDG_DATA_DIRS`). This can be fixed either on pressure-vessel side or freedesktop-sdk side.

**Comment by Taulim** [comment]
I see, It does make sense.

Did you do any modifications on your so it could work *out of the box*?
For now I just added this environment variable using flatseal and it seems to work.

Also, thank you for your time :)

**Comment by gasinvein** [comment]
> Did you do any modifications on your so it could work _out of the box_?

No, I just tested with Proton flatpak build. It uses runtime-provided vulkan-loader, which is patched too look for vulkan layers under the extensions mount point.

**Comment by gasinvein** [comment]
Since this issue is not fixable on the extension side, I think it belongs to the Steam flatpak app bugtracker.
